job-runner/runner/runner.go

```diff
@@ -39,33 +39,37 @@ type runner struct {
 	reqId   string    // the request id the job belongs to
 	rmc     rm.Client // client used to send JLs to the RM
 	// --
-	jobId     string
-	jobName   string
-	jobType   string
-	maxTries  uint
-	retryWait time.Duration
-	stopChan  chan struct{}
+	jobId       string
+	jobName     string
+	jobType     string
+	sequenceId  string
+	sequenceTry uint
+	maxTries    uint
+	retryWait   time.Duration
+	stopChan    chan struct{}
 	*sync.Mutex
 	logger    *log.Entry
 	startTime time.Time
 }
 
 // NewRunner takes a proto.Job struct and its corresponding job.Job interface, and
 // returns a Runner.
-func NewRunner(pJob proto.Job, realJob job.Job, reqId string, rmc rm.Client) Runner {
+func NewRunner(pJob proto.Job, realJob job.Job, reqId string, sequenceTry uint, rmc rm.Client) Runner {
```

Comment:
I think this begins to leak abstraction. I know we need seq id/try for the JLE, but imho job runner should not be aware of sequences, which are a higher abstraction. So in cases like this, "most problems can be solved by another level of abstraction". What we want is a thing to do logging, and we can give that thing all the info it needs (e.g. seq id/try), and then pass that thing into the job runner. So the job runner gets logging capabilities without any awareness of things it should be aware of.